sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -70,7 +75,9 @@ def _brute_mst(mutual_reachability, min_samples, sparse=False):
     # Compute the minimum spanning tree for the sparse graph
     sparse_min_spanning_tree = csgraph.minimum_spanning_tree(mutual_reachability)
     rows, cols = sparse_min_spanning_tree.nonzero()
-    return np.vstack((rows, cols, sparse_min_spanning_tree.data)).T
+    mst = np.vstack((rows, cols, sparse_min_spanning_tree.data))
+    mst = np.core.records.fromarrays(mst, dtype=MST_edge_dtype, shape=(mst.shape[1],))
```

Comment:
Do we need to vstack before calling `fromarrays`? In other words, can we do:

```suggestion
    mst = np.core.records.fromarrays(
        [rows, cols, sparse_min_spanning_tree.data],
        dtype=MST_edge_dtype)
```

Overall, this should remove a copy.